src/checkpoints.cpp

```diff
@@ -21,9 +21,9 @@ namespace Checkpoints {
         for (const MapCheckpoints::value_type& i : reverse_iterate(checkpoints))
         {
             const uint256& hash = i.second;
-            BlockMap::const_iterator t = mapBlockIndex.find(hash);
-            if (t != mapBlockIndex.end())
-                return t->second;
+            if (CBlockIndex* const pindex = LookupBlockIndex(hash)) {
```

Comment:
I dont think that is the type of const you meant. Probably just leave it non-const for now and I'll fix it up in #10692 (or you can start pulling stuff out of there into this PR if you prefer, but that probably just ends up taking more time).